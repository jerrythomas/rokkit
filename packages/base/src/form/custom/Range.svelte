<script>
	import RangeSlider from './RangeSlider.svelte'

	export let min
	export let max
	export let value = min
	export let step
	export let tickStep
	export let labelSkip

	let bounds = [min, value]

	$: value = bounds[1]
	$: console.log('range ', value)
</script>

<RangeSlider
	bind:value={bounds}
	{min}
	{max}
	{step}
	{tickStep}
	{labelSkip}
	single
/>
