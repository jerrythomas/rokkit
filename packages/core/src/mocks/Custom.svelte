<script>
	export let value
	export let fields
</script>

<span>{value[fields.text]}</span>
