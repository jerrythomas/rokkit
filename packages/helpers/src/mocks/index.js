import './animate'
import { ResizeObserver } from './resize-observer'

// skipcq: JS-E1004 - Needed for exposing all functions
export * from './match-media'
// skipcq: JS-E1004 - Needed for exposing all functions
export * from './element'

global.ResizeObserver = ResizeObserver
