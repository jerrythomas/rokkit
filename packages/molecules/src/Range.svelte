<script>
	import RangeMinMax from './RangeMinMax.svelte'

	/**
	 * @typedef {Object} Props
	 * @property {any} [name]
	 * @property {string} [class]
	 * @property {number} [min]
	 * @property {number} [max]
	 * @property {number} [value]
	 * @property {number} [step]
	 * @property {number} [ticks]
	 * @property {number} [labelSkip]
	 */

	/** @type {Props} */
	let {
		name = null,
		class: classes = '',
		min = 0,
		max = 100,
		value = $bindable(min),
		step = 1,
		ticks = 10,
		labelSkip = 0
	} = $props()

	let bounds = $state([min, value])

	$effect(() => {
		value = bounds[1]
	})
</script>

<RangeMinMax
	bind:value={bounds}
	{name}
	{min}
	{max}
	{step}
	{ticks}
	{labelSkip}
	single
	class={classes}
/>
