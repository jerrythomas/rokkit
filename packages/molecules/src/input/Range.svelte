<script>
	import RangeMinMax from './RangeMinMax.svelte'

	export let name = null
	/** @type {number} */
	export let min = 0
	/** @type {number} */
	export let max = 100
	/** @type {number} */
	export let value = min
	/** @type {number} */
	export let step = 1
	/** @type {number} */
	export let ticks = 10
	/** @type {number} */
	export let labelSkip = 0

	let bounds = [min, value]

	$: value = bounds[1]
</script>

<RangeMinMax bind:value={bounds} {name} {min} {max} {step} {ticks} {labelSkip} single />
