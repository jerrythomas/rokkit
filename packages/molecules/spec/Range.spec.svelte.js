import { beforeEach, describe, expect, it } from 'vitest'
import { cleanup, render } from '@testing-library/svelte'
import { flushSync } from 'svelte'
import Range from '../src/Range.svelte'
import '@rokkit/helpers/mocks'

describe('Range.svelte', () => {
	const ROOT_ELEMENT = 'rk-input-range'

	beforeEach(() => cleanup())
	it('should render', () => {
		const { container } = render(Range, {
			props: {
				name: 'range'
			}
		})
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()
	})
	it('should render with min, max', () => {
		const props = $state({
			name: 'range',
			min: 20,
			max: 40
		})
		const { container } = render(Range, { props })
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()
	})
	it('should render with ticks', () => {
		const props = $state({
			name: 'range',
			ticks: 5
		})
		const { container } = render(Range, { props })
		expect(container).toBeTruthy()
		expect(container).toMatchSnapshot()
	})

	it('should handle class changes', () => {
		const props = $state({
			name: 'range',
			class: 'test'
		})
		const { container } = render(Range, { props })
		expect(container).toMatchSnapshot()
		const range = container.querySelector(ROOT_ELEMENT)
		expect(Array.from(range.classList)).toContain('test')

		props.class = 'test2'
		flushSync()
		expect(container).toMatchSnapshot()
		expect(Array.from(range.classList)).toContain('test2')
	})

	it('should handle value changes', () => {
		const props = $state({
			name: 'range',
			value: 30
		})
		const { container } = render(Range, { props })
		expect(container).toMatchSnapshot()
		props.value = 20
		flushSync()
		expect(container).toMatchSnapshot()
	})
})
