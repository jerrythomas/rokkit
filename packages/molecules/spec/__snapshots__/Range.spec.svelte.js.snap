// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Range.svelte > should handle class changes 1`] = `
<div>
  <!---->
  <rk-input-range
    class="relative grid h-10 grid-rows-2 test"
  >
    <input
      name="range"
      type="hidden"
      value="0,0"
    />
     
    <rk-range-track
      class="relative grid svelte-n6pm5g"
    >
      <rk-range-track-bar
        class="relative col-start-2 box-border"
      />
       
      <rk-selected-bar
        class="absolute col-start-2"
        style="left: 0px; width: 0px;"
      />
       
      <!---->
       
      <rk-thumb
        aria-valuemax="0"
        aria-valuemin="0"
        aria-valuenow="0"
        class="absolute -top-1 box-border h-4 w-4 cursor-pointer"
        role="slider"
        style="left: 0px;"
        tabindex="0"
      />
      <!---->
    </rk-range-track>
     
    <rk-ticks
      class="svelte-n6pm5g"
      style="--count: 10;"
    >
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          0
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          10
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          20
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          30
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          40
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          50
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          60
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          70
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          80
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          90
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          100
        </p>
      </rk-tick>
      
    </rk-ticks>
  </rk-input-range>
  
  
</div>
`;

exports[`Range.svelte > should handle class changes 2`] = `
<div>
  <!---->
  <rk-input-range
    class="relative grid h-10 grid-rows-2 test2"
  >
    <input
      name="range"
      type="hidden"
      value="0,0"
    />
     
    <rk-range-track
      class="relative grid svelte-n6pm5g"
    >
      <rk-range-track-bar
        class="relative col-start-2 box-border"
      />
       
      <rk-selected-bar
        class="absolute col-start-2"
        style="left: 0px; width: 0px;"
      />
       
      <!---->
       
      <rk-thumb
        aria-valuemax="0"
        aria-valuemin="0"
        aria-valuenow="0"
        class="absolute -top-1 box-border h-4 w-4 cursor-pointer"
        role="slider"
        style="left: 0px;"
        tabindex="0"
      />
      <!---->
    </rk-range-track>
     
    <rk-ticks
      class="svelte-n6pm5g"
      style="--count: 10;"
    >
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          0
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          10
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          20
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          30
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          40
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          50
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          60
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          70
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          80
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          90
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          100
        </p>
      </rk-tick>
      
    </rk-ticks>
  </rk-input-range>
  
  
</div>
`;

exports[`Range.svelte > should handle value changes 1`] = `
<div>
  <!---->
  <rk-input-range
    class="relative grid h-10 grid-rows-2 "
  >
    <input
      name="range"
      type="hidden"
      value="0,30"
    />
     
    <rk-range-track
      class="relative grid svelte-n6pm5g"
    >
      <rk-range-track-bar
        class="relative col-start-2 box-border"
      />
       
      <rk-selected-bar
        class="absolute col-start-2"
        style="left: 0px; width: 0px;"
      />
       
      <!---->
       
      <rk-thumb
        aria-valuemax="0"
        aria-valuemin="0"
        aria-valuenow="30"
        class="absolute -top-1 box-border h-4 w-4 cursor-pointer"
        role="slider"
        style="left: 0px;"
        tabindex="0"
      />
      <!---->
    </rk-range-track>
     
    <rk-ticks
      class="svelte-n6pm5g"
      style="--count: 10;"
    >
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          0
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          10
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          20
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          30
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          40
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          50
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          60
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          70
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          80
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          90
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          100
        </p>
      </rk-tick>
      
    </rk-ticks>
  </rk-input-range>
  
  
</div>
`;

exports[`Range.svelte > should handle value changes 2`] = `
<div>
  <!---->
  <rk-input-range
    class="relative grid h-10 grid-rows-2 "
  >
    <input
      name="range"
      type="hidden"
      value="0,30"
    />
     
    <rk-range-track
      class="relative grid svelte-n6pm5g"
    >
      <rk-range-track-bar
        class="relative col-start-2 box-border"
      />
       
      <rk-selected-bar
        class="absolute col-start-2"
        style="left: 0px; width: 0px;"
      />
       
      <!---->
       
      <rk-thumb
        aria-valuemax="0"
        aria-valuemin="0"
        aria-valuenow="30"
        class="absolute -top-1 box-border h-4 w-4 cursor-pointer"
        role="slider"
        style="left: 0px;"
        tabindex="0"
      />
      <!---->
    </rk-range-track>
     
    <rk-ticks
      class="svelte-n6pm5g"
      style="--count: 10;"
    >
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          0
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          10
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          20
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          30
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          40
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          50
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          60
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          70
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          80
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          90
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          100
        </p>
      </rk-tick>
      
    </rk-ticks>
  </rk-input-range>
  
  
</div>
`;

exports[`Range.svelte > should render 1`] = `
<div>
  <!---->
  <rk-input-range
    class="relative grid h-10 grid-rows-2 "
  >
    <input
      name="range"
      type="hidden"
      value="0,0"
    />
     
    <rk-range-track
      class="relative grid svelte-n6pm5g"
    >
      <rk-range-track-bar
        class="relative col-start-2 box-border"
      />
       
      <rk-selected-bar
        class="absolute col-start-2"
        style="left: 0px; width: 0px;"
      />
       
      <!---->
       
      <rk-thumb
        aria-valuemax="0"
        aria-valuemin="0"
        aria-valuenow="0"
        class="absolute -top-1 box-border h-4 w-4 cursor-pointer"
        role="slider"
        style="left: 0px;"
        tabindex="0"
      />
      <!---->
    </rk-range-track>
     
    <rk-ticks
      class="svelte-n6pm5g"
      style="--count: 10;"
    >
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          0
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          10
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          20
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          30
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          40
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          50
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          60
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          70
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          80
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          90
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          100
        </p>
      </rk-tick>
      
    </rk-ticks>
  </rk-input-range>
  
  
</div>
`;

exports[`Range.svelte > should render with min, max 1`] = `
<div>
  <!---->
  <rk-input-range
    class="relative grid h-10 grid-rows-2 "
  >
    <input
      name="range"
      type="hidden"
      value="20,20"
    />
     
    <rk-range-track
      class="relative grid svelte-n6pm5g"
    >
      <rk-range-track-bar
        class="relative col-start-2 box-border"
      />
       
      <rk-selected-bar
        class="absolute col-start-2"
        style="left: 20px; width: 0px;"
      />
       
      <!---->
       
      <rk-thumb
        aria-valuemax="0"
        aria-valuemin="20"
        aria-valuenow="20"
        class="absolute -top-1 box-border h-4 w-4 cursor-pointer"
        role="slider"
        style="left: 20px;"
        tabindex="0"
      />
      <!---->
    </rk-range-track>
     
    <rk-ticks
      class="svelte-n6pm5g"
      style="--count: 10;"
    >
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          20
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          22
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          24
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          26
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          28
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          30
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          32
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          34
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          36
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          38
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          40
        </p>
      </rk-tick>
      
    </rk-ticks>
  </rk-input-range>
  
  
</div>
`;

exports[`Range.svelte > should render with ticks 1`] = `
<div>
  <!---->
  <rk-input-range
    class="relative grid h-10 grid-rows-2 "
  >
    <input
      name="range"
      type="hidden"
      value="0,0"
    />
     
    <rk-range-track
      class="relative grid svelte-n6pm5g"
    >
      <rk-range-track-bar
        class="relative col-start-2 box-border"
      />
       
      <rk-selected-bar
        class="absolute col-start-2"
        style="left: 0px; width: 0px;"
      />
       
      <!---->
       
      <rk-thumb
        aria-valuemax="0"
        aria-valuemin="0"
        aria-valuenow="0"
        class="absolute -top-1 box-border h-4 w-4 cursor-pointer"
        role="slider"
        style="left: 0px;"
        tabindex="0"
      />
      <!---->
    </rk-range-track>
     
    <rk-ticks
      class="svelte-n6pm5g"
      style="--count: 5;"
    >
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          0
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          20
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          40
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          60
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          80
        </p>
      </rk-tick>
      <rk-tick
        aria-selected="false"
        class="grid cursor-pointer select-none grid-cols-2"
        role="option"
        tabindex="0"
      >
        <rk-tick-bar
          class="h-5px col-start-2 border-l"
        />
         
        <p
          class="col-span-2 flex justify-center"
        >
          100
        </p>
      </rk-tick>
      
    </rk-ticks>
  </rk-input-range>
  
  
</div>
`;
